src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -102,6 +102,22 @@ class FunctionDefinitionGrammar(BaseSegment):
     )
 
 
+@postgres_dialect.segment(replace=True)
```

Comment:
Is it possible we want to modify `FunctionParameterListGrammar` in the base (ANSI) dialect rather than making this Postgres-specific?

Reply:
Yes sure, I will move it. I decided to make it postgres centric because I haven't worked with other dialect and not sure it they also support zero argument.

Reply:
I'm not certain about other databases, but it seems very likely. Also, we tend to have a flexible, permissive approach to the ANSI dialect if it helps us keep the overall code simpler. I hope that makes sense! We may formalize and document this philosophy someday.

Reply:
Yeah, it makes lot of sense. Since it's going to be flexible, I don't expect it to break the other databases.